src/api/auth-identities.tsx

```diff
@@ -0,0 +1,109 @@
+import { handleResponse, handleSettledResult } from "util/helpers";
+import { LxdApiResponse } from "types/apiResponse";
+import { LxdGroup, LxdIdentity } from "types/permissions";
+import { getIdentityIdsForGroup } from "util/permissionIdentities";
+
+export const fetchIdentities = (): Promise<LxdIdentity[]> => {
+  return new Promise((resolve, reject) => {
+    fetch(`/1.0/auth/identities?recursion=1`)
+      .then(handleResponse)
+      .then((data: LxdApiResponse<LxdIdentity[]>) => resolve(data.metadata))
+      .catch(reject);
+  });
+};
+
+export const fetchIdentity = (
+  id: string,
+  authMethod: string,
+): Promise<LxdIdentity> => {
+  return new Promise((resolve, reject) => {
+    fetch(`/1.0/auth/identities/${authMethod}/${id}`)
+      .then(handleResponse)
+      .then((data: LxdApiResponse<LxdIdentity>) => resolve(data.metadata))
+      .catch(reject);
+  });
+};
+
+export const updateGroupsForIdentity = (
+  identity: Partial<LxdIdentity>,
+  groups: string[],
+): Promise<void> => {
+  return new Promise((resolve, reject) => {
+    fetch(
+      `/1.0/auth/identities/${identity.authentication_method}/${identity.id}`,
+      {
+        method: "PUT",
+        body: JSON.stringify({
+          groups,
+        }),
+      },
+    )
+      .then(handleResponse)
+      .then(resolve)
+      .catch(reject);
+  });
+};
+
+export const updateGroupsForIdentities = (
+  identities: Partial<LxdIdentity>[],
+  groupsForIdentities: Record<string, string[]>,
+): Promise<void> => {
+  return new Promise((resolve, reject) => {
+    void Promise.allSettled(
+      identities.map((identity) =>
+        updateGroupsForIdentity(
+          identity,
+          groupsForIdentities[identity.id ?? ""] || [],
+        ),
+      ),
+    )
+      .then(handleSettledResult)
+      .then(resolve)
+      .catch(reject);
+  });
+};
```

Comment:
I think it would be more straight forward to call those `updateIdentity` and pass in the identity object with the updated groups. Then we wouldn't need a 2nd array `groupsForIdentities` in the batch case with matching entries, which is a bit indirect concept to have.

Reply:
agreed!